lib/ansible_tower_client/job_template.rb

```diff
@@ -10,11 +10,15 @@ def initialize(api, raw_body)
       super
     end
 
-    def launch(vars = {})
+    def launch(options = {})
       launch_url = "#{url}launch/"
-      extra = JSONValues.new(vars).extra_vars
-      resp = api.post(launch_url, extra).body
-      job = JSON.parse(resp)
+      new_limit = options.dup.delete('limit')
+      response = with_temporary_changes(new_limit) do
+        api.post(launch_url, options).body
+      end
+
+      job = JSON.parse(response)
+      raise InvalidOptions.new(self, Error.parse(response)) if job['job'].nil?
```

Comment:
Thanks, the discussion cleared up a lot.  You're correct, this is a different problem from the resource not being found.
I was expecting Faraday to raise an error when it receives a HTTP 400 response from the `api.post` above.  I think we should probably handle this error generically in the Api class.  Let's do that in a followup PR.